bliss/models/galaxy_net.py

```diff
@@ -139,43 +140,68 @@ def validation_step(self, batch, batch_idx):  # pylint: disable=unused-argument
         loss = self.get_loss(images, recon_mean)
 
         # metrics
+        # FIXME: change to val/loss so it's automatically grouped in tensorboard
         self.log("val_loss", loss)
         residuals = (images - recon_mean) / torch.sqrt(images)
         self.log("val_max_residual", residuals.abs().max())
         return {"images": images, "recon_mean": recon_mean}
 
     def validation_epoch_end(self, outputs):
-        images = outputs[0]["images"][:10]
-        recon_mean = outputs[0]["recon_mean"][:10]
+        images = torch.cat([x["images"] for x in outputs])
+        recon_mean = torch.cat([x["recon_mean"] for x in outputs])
         fig = self.plot_reconstruction(images, recon_mean)
         if self.logger:
             self.logger.experiment.add_figure(f"Images {self.current_epoch}", fig)
 
     def plot_reconstruction(self, images, recon_mean):
-        # only plot i band if available, otherwise the highest band given.
+
+        # 1. only plot i band if available, otherwise the highest band given.
+        # 2. plot the `num_examples` of images with the largest average residual
+        #    across all batches in the last epoch
+        # 3. residual color range (`vmin`, `vmax`) are fixed as across all samples
```

Comment:
It might be good to mention here that you fix them across all images to be the maximum/minimum value across all images. (right?)